classes/submission/reviewAssignment/Repository.php

```diff
@@ -0,0 +1,256 @@
+     *
+     * @param array $props A key/value array with the new data to validate
+     * @param array $allowedLocales The context's supported locales
+     * @param string $primaryLocale The context's primary locale
+     *
+     * @return array A key/value array with validation errors. Empty if no errors
+     */
+    public function validate(?ReviewAssignment $object, array $props, Context $context): array
+    {
+        $primaryLocale = $context->getData('primaryLocale');
+        $allowedLocales = $context->getData('supportedFormLocales');
+
+        $validator = ValidatorFactory::make(
+            $props,
+            $this->schemaService->getValidationRules($this->dao->schema, $allowedLocales),
+            []
+        );
+
+        // Check required fields
+        ValidatorFactory::required(
+            $validator,
+            $object,
+            $this->schemaService->getRequiredProps($this->dao->schema),
+            $this->schemaService->getMultilingualProps($this->dao->schema),
+            $allowedLocales,
+            $primaryLocale
+        );
+
+        // Check if submission exists
+        if (isset($props['submissionId'])) {
+            $validator->after(function($validator) use ($props) {
+                if (!$validator->errors()->get('submissionId')) {
+                    $submission = Repo::submission()->get($props['submissionId']);
+                    if (!$submission) {
+                        $validator->errors()->add('submissionId', __('api.reviews.assignments.invalidSubmission'));
+                    }
+                }
+            });
+        }
+
+        // Check if reviewer exists
+        if (isset($props['reviewerId'])) {
+            $validator->after(function($validator) use ($props, $context) {
+                if (!$validator->errors()->get('reviewerId')) {
+                    $reviewer = Repo::user()->get($props['reviewerId']);
+                    if (!$reviewer) {
+                        $validator->errors()->add('reviewerId', __('api.reviews.assignments.invalidReviewer'));
+                    }
+                    $roleDao = DAORegistry::getDAO('RoleDAO'); /** @var RoleDAO $roleDao */
+                    $isReviewer = $roleDao->userHasRole($context->getId(), $reviewer->getId(), Role::ROLE_ID_REVIEWER);
+                    if (!$isReviewer) {
+                        $validator->errors()->add('reviewerId', __('api.reviews.assignments.invalidReviewer'));
+                    }
+                }
+            });
+        }
+
+        // Check for input from disallowed locales
+        ValidatorFactory::allowedLocales($validator, $this->schemaService->getMultilingualProps($this->dao->schema), $allowedLocales);
+
+        $errors = [];
+
+        if ($validator->fails()) {
+            $errors = $this->schemaService->formatValidationErrors($validator->errors());
+        }
+
+        Hook::call('Category::validate', [&$errors, $object, $props, $allowedLocales, $primaryLocale]);
```

Comment:
Wrong hook name.

Reply:
Sorry, (hopefully) one more request related to hooks :) Can you document the hooks that got removed in the issue, along with anything else that should go into the Release Notebook, and then drop a link to that comment into https://github.com/pkp/pkp-lib/issues/9276 where we track these things?

Reply:
One more :) When this code gets linted, you should see the pre-commit hook come in to add hook self-documentation in the function self-docs for the new hooks. If that doesn't happen, can you check whether your pre-commit hooks are running as expected?